providers/implementations/rands/seeding/rand_cpu_x86.c

```diff
@@ -57,6 +54,53 @@ size_t prov_acquire_entropy_from_cpu(RAND_POOL *pool)
 
     return rand_pool_entropy_available(pool);
 }
+
+#if defined(OPENSSL_SYSNAME_TANDEM) && defined(_TNS_X_TARGET)
+/* Obtain random bytes from the x86 hardware random function in 64 bit chunks */
+static size_t get_hardware_random_value(unsigned char *buf, size_t len)
+{
+    size_t bytes_remaining = len;
+
+    while (bytes_remaining > 0) {
+        /* Always use 64 bit fetch, then use the lower bytes as needed. */
+        /* The platform is big-endian. */
+        unsigned long long random_value = 0;
+
+        if (_rdrand64(&random_value) != 0) {
```

Comment:
I can't find any manual about _rdrand64, but does it take an unsigned long long, or an uint64_t?

Reply:
> 
> 
> I can't find any manual about _rdrand64, but does it take an unsigned long long, or an uint64_t?

The method is a builtin intrinsic function on the HPE NonStop TNS/X platform. It is documented in their operating system API documentation site only, which is not SEO optimized. It takes a `uint64_t *` (which the compiler is not defining for some reason, but that is equivalent to `unsigned long long *` on the platform).

Reply:
We use uint64_t in various files, so it really should be available.

Reply:
Pushed a new change using uint64_t. Probably operator error last time.